ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -762,7 +762,11 @@ let do_monitor_write xc writers =
       let stats = List.rev_append plugins_stats dom0_stats in
       Rrdd_stats.print_snapshot () ;
       let uuid_domids = List.map (fun (_, u, i) -> (u, i)) domains in
-      Rrdd_monitor.update_rrds timestamp stats uuid_domids my_paused_vms
+      Rrdd_monitor.update_rrds timestamp stats uuid_domids my_paused_vms ;
+
+      (* Dump to /dev/shm/metrics/host-dss *)
+      Rrdd_server.Plugin.get_path "host-dss"
+      |> Rrdd_server.dump_host_dss_to_file
```

Comment:
Is it possible for other readers (like NRPE) conflicting with rrdd on accessing the file, like reading corrupted content?

Reply:
The reader takes responsibility to retry. I have updated design https://info.citrite.net/display/xenserver/Design+of+NRPE+integration

Reply:
Then how could a reader know that corrupted data is being read?

Reply:
I will change to that.

Reply:
Fixed now by referring `Unixext.atomic_write_to_file`

Reply:
I seems you can use `Unixext.atomic_write_to_file` directly.
The `fd` passed from it can be changed to channel by `Unix.out_channel_of_descr`.
https://v2.ocaml.org/api/Unix.html

Reply:
It's wonderful!, fixed now!